### PR TITLE
[#169127957] added profile info in central header

### PR DESCRIPTION
**Pull request by d-al-ibm**
added user name and role in central header
added user profile management in dashboard and default container
added new translations and mock services

### CODE REVIEW

File: src/components/CentralHeader/CentralHeader.tsx [hunk 1]
```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
...
   /**
    * react-i18next translation hook
    */
   const { t } = useTranslation();
 
+  /**
+   * add user icon if user name is set
+   */
+  const userIcon =
+    props.userName !== "" && props.userName !== " " ? (
```
Comment (gunzip) [inline src/components/CentralHeader/CentralHeader.tsx]: why `userName` could be "" or " " ?

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#169127957](https://www.pivotaltracker.com/story/show/169127957): Gestione informazioni topbar

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e3d2372b9aa4e0dc206c3471cb1fc5039bf7afb3
</p>

File: src/components/CentralHeader/CentralHeader.tsx (hunk 1 above)
Comment (d-al-ibm) [reply src/components/CentralHeader/CentralHeader.tsx]: as union of two strings from the getProfile service (family_name and given_name), I had to join these strings with a whitespace and check for the " "; now i substituted the check with string.trim() to join both checks

File: src/components/CentralHeader/CentralHeader.tsx [hunk 2]
```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
...
+export const CentralHeader = withRouter((props: ICentralHeaderProps) => {
   /**
    * react-i18next translation hook
    */
   const { t } = useTranslation();
 
+  /**
+   * add user icon if user name is set
+   */
+  const userIcon =
```
Comment (gunzip) [inline src/components/CentralHeader/CentralHeader.tsx]: consider also to make a `<UserIcon >` component that takes the username as input prop

File: src/components/CentralHeader/CentralHeader.tsx [hunk 3]
```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
...
+      </svg>
+    ) : null;
+
+  /**
+   * switch from user role enum to string
+   */
+  const userRole = (userRoleEnum => {
+    switch (userRoleEnum) {
+      case UserRoleEnum.ORG_DELEGATE:
+        return t("centralHeader.userBadge.delegate");
```
Comment (gunzip) [inline src/components/CentralHeader/CentralHeader.tsx]: I'm a little bit skeptical about the naming of the locales. If these values are supposed to be used somewhere else is better to drop the prefix "centralHeader" and name them with some domain concept (ie. `user.badge.delegate`)

File: src/components/CentralHeader/CentralHeader.tsx [hunk 4]
```diff
@@ -31,22 +32,42 @@ interface ICentralHeaderProps extends RouteComponentProps {
...
+    }
+  })(props.userRole);
   /**
    * create badge for user role if user role is set
    */
   const userRoleBadge =
-    props.userRole !== "" ? (
-      <Badge color="white">{props.userRole}</Badge>
-    ) : null;
+    props.userRole !== "" ? <Badge color="white">{userRole}</Badge> : null;
```
Comment (gunzip) [inline src/components/CentralHeader/CentralHeader.tsx]: consider to make a component `<UserRoleBadge >` that takes userRole as input prop (we also have a NonEmptyString type if you dare : )

File: src/components/CentralHeader/CentralHeader.tsx (hunk 2 above)
Comment (d-al-ibm) [reply src/components/CentralHeader/CentralHeader.tsx]: I see the advantage of creating child components for a more ordered code and to give each component it's logic, but in this case the amount of code is quite small...which would be the advantage of creating a component for such few lines of code?

File: src/components/CentralHeader/CentralHeader.tsx (hunk 4 above)
Comment (d-al-ibm) [reply src/components/CentralHeader/CentralHeader.tsx]: I see the advantage of creating child components for a more ordered code and to give each component it's logic, but in this case the amount of code is quite small...which would be the advantage of creating a component for such few lines of code?
about NonEmptyString class, I only find NonEmptyArray in fp-ts...where can I find it?

File: src/components/CentralHeader/CentralHeader.tsx (hunk 3 above)
Comment (d-al-ibm) [reply src/components/CentralHeader/CentralHeader.tsx]: I agree with you but I would prefer to make these changes in a separate pr (I created https://www.pivotaltracker.com/story/show/169342054 for this purpose)

File: src/components/CentralHeader/CentralHeader.tsx (hunk 2 above)
Comment (gunzip) [reply src/components/CentralHeader/CentralHeader.tsx]: I see the same advantage of breaking code into smaller pieces: it favors testability (think about: how would you test these lines here ?) composability and reusability (what if I have to put an userIcon somewhere else and then change appearance ?). Now you have a big component (CentralHeader) that spans 150 lines of code. I'm not saying you have to create another file, just put these lines outside `CentralHeader` in the same source; you'll end up with a shorter method and your code will be more declarative. While this suggestion may be negligibile in this specific context, it is a general advice : -)

File: src/components/CentralHeader/CentralHeader.tsx [hunk 5]
```diff
@@ -14,80 +14,119 @@ import {
...
   /**
    * function to navigate to profile
    */
   const navigateToProfile = () => props.history.push("/profile");
 
+  const userRoleBadge = NonEmptyString.decode(props.userRole).isRight() ? (
+    <UserRoleBadge userRole={props.userRole} />
+  ) : null;
+
+  const userIcon = NonEmptyString.decode(props.userName.trim()).isRight() ? (
```
Comment (gunzip) [inline src/components/CentralHeader/CentralHeader.tsx]: shorter: 

`const userIcon = NonEmptyString.is(props.userName.trim()) ? ...`

or 

```ts
const userIcon = NonEmptyString.decode(props.userName.trim()).fold(
   _ => null,
   _ =><UserIcon />
);
```

File: src/context/token-context.tsx [hunk 6]
```diff
@@ -12,14 +12,14 @@ export const TokenContext = createContext<ITokenContext>({
   setToken: () => {
     return;
```
Comment (gunzip) [inline src/context/token-context.tsx]: is this is a stub please add a comment

**Pull request merged**